src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java

```diff
@@ -457,7 +457,7 @@ protected LayoutManager createLayoutManager() {
     class AquaComboBoxLayoutManager extends BasicComboBoxUI.ComboBoxLayoutManager {
         protected Rectangle rectangleForCurrentValue() {
             int width = comboBox.getWidth();
-            int height = 22;
+            int height = comboBox.getBorder() == null ? 22 : comboBox.getHeight();
```

Comment:
This looks to me like it will need reworking in future - "22" seems a bit too much of a magic number - but reverting to using the height for the problem case seems safe.
However please do run all related Swing tests on Aqua INCLUDING manual tests of the combo-box - perhaps using SwingSet2 - to make sure it looks good.